api/v1beta1/clusterclass_types.go

```diff
@@ -583,6 +583,42 @@ type JSONSchemaProps struct {
 	// It can be used to add additional data for higher level tools.
 	// +optional
 	XMetadata *VariableSchemaMetadata `json:"x-metadata,omitempty"`
+
+	// AllOf specifies that the variable must validate against all of the subschemas in the array.
+	// NOTE: Can only be set if type is object.
```

Comment:
Are we sure this is only allowed for type object? (is there some evidence for it either in the upstream k/k implementation or in OpenAPI Schema v3?)

If yes, I assume it's allowed for "regular" objects and maps? (both have type object)

If it's true, let's please validate that these fields are only set if type is object (in clusterclass_variable_validation.go validateSchema)

Reply:
Hmm I think this is a copy-paste error, should be applicable to all types.